documentation/design/software_design.md

```diff
@@ -0,0 +1,39 @@
+# **Gitops Plugin Design**
```

Comment:
General comments:
     Use present tense in all statement.
     Mention the data structures used
     This design gives Introduction to components.  The actions performed by the component has to be mentioned in separate bullet list.  Though it looks like a low level design, this being a small component, those details are acceptable. If one goes through these statements, it should be clear about the sequence of statements/actions performed by the component.
     This design does not cover the execution flow. For example, if there are 10 jobs, how these are scheduled, and how they get executed.
     Error cases have to be thought through and has to be captured in design.

Reply:
These are the additional comments.
How to stop the system? When a jobs are in progress, these have to be gracefully completed.